scripts/setup-adapters.sh

```diff
@@ -16,44 +16,28 @@
 # Propagate errors and improve debugging.
 set -eufx -o pipefail
 
+SCRIPTDIR=$(dirname "$0")
+source $SCRIPTDIR/setup-helper-functions.sh
+DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}
+
 function install_aws-sdk-cpp {
-  local NAME="aws-sdk-cpp"
+  local AWS_REPO_NAME="aws/aws-sdk-cpp"
   local AWS_SDK_VERSION="1.9.96"
 
-  if [ -d "${NAME}" ]; then
-    read -r -p "do you want to rebuild '${NAME}'? (y/n) " confirm
-    if [[ "${confirm}" =~ ^[yy]$ ]]; then
-      rm -rf "${NAME}"
-    else
-      return 0
-    fi
-  fi
-
-  git clone --depth 1 --recurse-submodules --branch "${AWS_SDK_VERSION}" https://github.com/aws/aws-sdk-cpp.git "${NAME}"
-  mkdir "${NAME}_build"
-  cd "${NAME}_build" || exit
-  cmake ../"${NAME}" \
-    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
-    -DCMAKE_BUILD_TYPE=Debug \
-    -DBUILD_ONLY:STRING="s3;identity-management" \
-    -DBUILD_SHARED_LIBS:BOOL=OFF \
-    -DMINIMIZE_SIZE:BOOL=ON \
-    -DENABLE_TESTING:BOOL=OFF \
-    -DCMAKE_INSTALL_PREFIX="${DEPENDENCY_DIR}/install" \
-    -GNinja \
-    .
-  local _ret=$?
-  if [ $_ret -ne 0 ] ; then
-     echo "cmake returned with exit code $_ret, aborting!" >&2
-     return $_ret
-  fi
-  ninja
-  ninja install
+  github_checkout $AWS_REPO_NAME $AWS_SDK_VERSION --depth 1 --recurse-submodules
+  cmake_install -DCMAKE_BUILD_TYPE=Debug -DBUILD_SHARED_LIBS:BOOL=OFF -DMINIMIZE_SIZE:BOOL=ON -DENABLE_TESTING:BOOL=OF
 }
 
 cd "${DEPENDENCY_DIR}" || exit
 # aws-sdk-cpp missing dependencies
-yum install -y curl-devel
+
+if [[ "$OSTYPE" == linux-gnu* ]]; then
+   yum install -y curl-devel
```

Comment:
so question her actually, I realized that all of our scripts are using apt to install packages and here we have yum. 

Those are two different distributions of Linux. Should I change this to apt? Everywhere else uses apt as far as I can tell.

Reply:
sure.

Reply:
CI uses dnf which is rhel based. rhel also uses yum?

Reply:
I see this in the circle ci output:
Build-agent version 1.0.123398-a9b2faa6 (2022-05-12T19:23:54+0000)
System information:
 Server Version: 20.10.14
 Storage Driver: overlay2
  Backing Filesystem: xfs
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Kernel Version: 5.13.0-1021-aws
 Operating System: Ubuntu 20.04.4 LTS
 OSType: linux
 Architecture: x86_64

but yum does work there

Reply:
I maybe can just provide yum and apt and let one command fail?

Reply:
curl install should ideally be removed and should be baked into the image. Do you need apt for some use case?

Reply:
curl is baked in the Velox CI image. But this script is also used for presto_cpp where I am not sure if curl is baked.

Reply:
setup adapters would be run locally as well? If you need to develop for s3 or hdfs for example youll have to run it locally, we have setup-ubuntu so im asuming locally you would be running:

setup-ubuntu
setup-adapters

Shall I remove curl install?

Reply:
Looks like presto_cpp CI image now has curl installed also. So this can be fully removed.

`Package libcurl-devel-7.61.1-22.el8.x86_64 is already installed.`
https://circleci.com/api/v1.1/project/github/facebookexternal/presto_cpp/19176/output/105/0?file=true&allocation-id=627d7088dec82e3918d27646-0-build%2FD450F00

Same with Velox 
https://circleci.com/api/v1.1/project/github/facebookincubator/velox/44004/output/105/0?file=true&allocation-id=627d687cdec82e3918d245f8-0-build%2F568F9045

Reply:
> Shall I remove curl install?

yes please.